client/signup/steps/domains/index.jsx

```diff
@@ -244,12 +245,25 @@ export class RenderDomainsStep extends Component {
 		}
 	};
 
-	handleDomainMappingError = ( domain_name ) => {
+	handleDomainMappingError = async ( domain_name ) => {
 		const productToRemove = this.props.cart.products.find(
 			( product ) => product.meta === domain_name
 		);
+
 		if ( productToRemove ) {
-			this.removeDomain( { domain_name, product_slug: productToRemove.product_slug } );
+			this.setState( ( prevState ) => ( {
```

Comment:
Here we're ensuring multiple requests for auto-deletion on failed domains will work as expected by removing them.